src/main/java/org/radarcns/auth/RadarSecurityContext.java

```diff
@@ -0,0 +1,53 @@
+package org.radarcns.auth;
+
+import java.security.Principal;
+import java.util.Collections;
+import javax.ws.rs.core.SecurityContext;
+import org.radarcns.auth.token.RadarToken;
+
+/**
+ * Security context from a {@link RadarToken}.
+ */
+public class RadarSecurityContext implements SecurityContext {
+    private final RadarToken token;
+
+    /** Constructs a SecurityContext from a RadarToken. */
+    public RadarSecurityContext(RadarToken token) {
+        this.token = token;
+    }
+
+    @Override
+    public Principal getUserPrincipal() {
+        return token::getSubject;
+    }
+
+    /**
+     * Maps roles in the shape {@code "project:role"} to a Management Portal role. Global roles
+     * take the shape of {@code ":global_role"}.
+     * @param role role to be mapped
+     * @return {@code true} if the RadarToken contains given project/role,
+     *         {@code false} otherwise
+     */
+    @Override
+    public boolean isUserInRole(String role) {
```

Comment:
Can you please explain this a bit more?

Reply:
It is just an extension of the standard Jersey SecurityContext. It allows annotations like
`@RolesAllowed(":SYS_ADMIN")`, without further detailed checking of the `RadarToken`. See https://jersey.github.io/documentation/latest/security.html

Reply:
Thanks. Did not know that !